src/model/wrapper.jl

```diff
@@ -61,128 +36,74 @@ end
+    },
 ) where {T} = true
 
-function MOI.get(
-    model::VirtualQUBOModel,
-    attr::Union{
-        MOI.SolveTimeSec,
-        MOI.PrimalStatus,
-        MOI.DualStatus,
-        MOI.TerminationStatus,
-        MOI.RawStatusString,
-    }
-)
-    if !isnothing(model.optimizer)
-        MOI.get(model.optimizer, attr)
-    else
-        nothing
-    end
-end
 
-function MOI.set(
-    model::VirtualQUBOModel,
-    attr::Union{
-        MOI.SolveTimeSec,
-        MOI.PrimalStatus,
-        MOI.DualStatus,
-        MOI.TerminationStatus,
-        MOI.RawStatusString,
-    },
-    value::Any,
-)
-    return MOI.set(model.optimizer, attr, value)
-end
+PBO.showvar(x::VI) = PBO.showvar(x.value)
 
-MOI.supports(::VirtualQUBOModel, ::MOI.SolveTimeSec) = true
+PBO.varlt(x::VI, y::VI) = PBO.varlt(x.value, y.value)
 
-function MOI.get(model::VirtualQUBOModel, rc::MOI.ResultCount)
-    if isnothing(model.optimizer)
-        return 0
-    else
-        return MOI.get(model.optimizer, rc)
-    end
-end
+function PBO.varlt(x::Set{V}, y::Set{V}) where {V}
+    if length(x) == length(y)
+        xv = sort!(collect(x); lt = PBO.varlt)
+        yv = sort!(collect(y); lt = PBO.varlt)
 
-MOI.supports(::VirtualQUBOModel, ::MOI.ResultCount) = true
+        for (xi, yi) in zip(xv, yv)
+            if xi == yi
+                continue
+            else
+                return PBO.varlt(xi, yi)
+            end
+        end
 
-function MOI.get(model::VirtualQUBOModel{T}, ov::MOI.ObjectiveValue) where {T}
-    if isnothing(model.optimizer)
-        return zero(T)
+        return false
     else
-        return MOI.get(model.optimizer, ov)
+        return length(x) < length(y)
     end
 end
 
-function MOI.get(model::VirtualQUBOModel{T}, vp::MOI.VariablePrimal, x::VI) where {T}
-    if isnothing(model.optimizer)
-        return zero(T)
-    else
-        v = MOI.get(model, Source(), x)
-        s = zero(T)
+const Optimizer{T} = VirtualModel{T}
 
-        for (ω, c) in expansion(v)
-            for y in ω
-                c *= MOI.get(model.optimizer, vp, y)
-            end
-            
-            s += c
-        end
-        
-        return s
-    end
-end
-
-MOI.get(::VirtualQUBOModel, ::MOI.SolverName)       = "Virtual QUBO Model"
-MOI.get(::VirtualQUBOModel, ::MOI.SolverVersion)    = PROJECT_VERSION
+# -*- QUBOTools -*- #
+function qubo(model, type::Type = Dict)
+    n, L, Q, α, β = MOI.get(model, QUBOTOOLS_NORMAL_FORM())
 
-function MOI.get(model::VirtualQUBOModel, rs::MOI.RawSolver)
-    if isnothing(model.optimizer)
-        return nothing
-    else
-        return MOI.get(model.optimizer, rs)
-    end
+    return QUBOTools.qubo(type, n, L, Q, α, β)
 end
 
-PBO.showvar(x::VI)       = PBO.showvar(x.value)
-PBO.varcmp(x::VI, y::VI) = PBO.varcmp(x.value, y.value)
+function ising(model, type::Type = Dict)
+    n, L̄, Q̄, ᾱ, β̄ = MOI.get(model, QUBOTOOLS_NORMAL_FORM())
+    L, Q, α, β    = QUBOTools.cast(QUBOTools.𝔹, QUBOTools.𝕊, L̄, Q̄, ᾱ, β̄)
```

Comment:
A little abusive with notation but if this makes you happy, I'm happy

Reply:
It is good that you mentioned it. I'm not very happy with our current interface between QT and TQ/Anneal. I wish we could redesign it soon.

Reply:
Basically, this things did not integrate as I expected (even though things got really better after models and formats became distinct entities) and also there are a few things that are implemented in both packages (TQ & Anneal) just because QT doens't import MOI.

Reply:
This is related to the PR that you want to put together for the MOI function migration to QUBOTools. Create an Issue for this, but we will leave it open instead of having it block this PR